<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>673</width>
    <height>517</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>10</number>
      </property>
      <item>
       <widget class="QComboBox" name="tableSelect">
        <property name="statusTip">
         <string>Table Select</string>
        </property>
        <property name="sizeAdjustPolicy">
         <enum>QComboBox::AdjustToMinimumContentsLength</enum>
        </property>
        <property name="minimumContentsLength">
         <number>20</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="SQLTextEdit" name="sqlLine">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>24</height>
         </size>
        </property>
        <property name="toolTip">
         <string>SQL Where Statement</string>
        </property>
        <property name="statusTip">
         <string>SQL Where Statement</string>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="lineWrapMode">
         <enum>QPlainTextEdit::NoWrap</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="buttonAssist">
        <property name="text">
         <string>Assist</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="buttonClear">
        <property name="text">
         <string>Clear</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QTableView" name="tableView_2">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <property name="verticalScrollMode">
        <enum>QAbstractItemView::ScrollPerPixel</enum>
       </property>
       <property name="horizontalScrollMode">
        <enum>QAbstractItemView::ScrollPerPixel</enum>
       </property>
       <property name="wordWrap">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QTableView" name="tableView">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <property name="verticalScrollMode">
        <enum>QAbstractItemView::ScrollPerPixel</enum>
       </property>
       <property name="horizontalScrollMode">
        <enum>QAbstractItemView::ScrollPerPixel</enum>
       </property>
       <property name="wordWrap">
        <bool>false</bool>
       </property>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QTabBar" name="tabView" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="statusLabel">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>0</height>
         </size>
        </property>
        <property name="autoFillBackground">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>No Selection</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Minimum</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>10</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>673</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <widget class="QMenu" name="menuRecent_Files">
     <property name="title">
      <string>Recent Files</string>
     </property>
     <addaction name="actionClear"/>
    </widget>
    <addaction name="actionConnect_to_database"/>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="menuRecent_Files"/>
    <addaction name="actionOpen_In_Memory_Database"/>
    <addaction name="separator"/>
    <addaction name="actionDuplicate_connection"/>
    <addaction name="actionDatabase_Information"/>
    <addaction name="separator"/>
    <addaction name="actionView_in_File_Manager"/>
    <addaction name="actionClose"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuTable">
    <property name="title">
     <string>Table</string>
    </property>
    <widget class="QMenu" name="menuShowHiddenColumn">
     <property name="title">
      <string>Show hidden column</string>
     </property>
     <addaction name="actionShowDummyHiddenColumn"/>
    </widget>
    <addaction name="actionCreateTable"/>
    <addaction name="actionImportTable"/>
    <addaction name="actionExport_Table"/>
    <addaction name="actionDrop_Table"/>
    <addaction name="actionDuplicate_Table"/>
    <addaction name="actionAttach_Database"/>
    <addaction name="separator"/>
    <addaction name="actionInsert"/>
    <addaction name="actionDelete"/>
    <addaction name="actionRefresh"/>
    <addaction name="actionRun_Custum_SQL"/>
    <addaction name="separator"/>
    <addaction name="menuShowHiddenColumn"/>
    <addaction name="separator"/>
    <addaction name="actionCommit"/>
    <addaction name="actionRevert"/>
    <addaction name="separator"/>
    <addaction name="actionPlot"/>
    <addaction name="actionR_code_to_import"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout_Table_View"/>
    <addaction name="actionAbout_Qt"/>
    <addaction name="separator"/>
    <addaction name="actionGo_github"/>
    <addaction name="actionGo_to_SQLite3_webpage"/>
    <addaction name="actionGo_to_Hyokai_info"/>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="title">
     <string>Window</string>
    </property>
    <addaction name="actionDummy"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionCopy"/>
    <addaction name="actionCopy_with_header"/>
    <addaction name="separator"/>
    <addaction name="actionSelect_All"/>
    <addaction name="separator"/>
    <addaction name="actionPreference"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>&amp;Show</string>
    </property>
    <addaction name="actionUse_fixed_width_font"/>
    <addaction name="separator"/>
    <addaction name="actionShow_Toolbar"/>
    <addaction name="actionShow_Table_List_View"/>
    <addaction name="actionShow_Column_List_View"/>
    <addaction name="actionSplit_Window"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuTable"/>
   <addaction name="menuWindow"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="windowTitle">
    <string>Main Toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionCreateTable"/>
   <addaction name="actionDrop_Table"/>
   <addaction name="actionRefresh"/>
   <addaction name="actionRun_Custum_SQL"/>
   <addaction name="actionImportTable"/>
   <addaction name="actionExport_Table"/>
   <addaction name="actionPlot"/>
   <addaction name="actionInsert"/>
   <addaction name="actionDelete"/>
   <addaction name="actionCommit"/>
   <addaction name="actionRevert"/>
  </widget>
  <widget class="QDockWidget" name="tableListWidget">
   <property name="windowTitle">
    <string>Table List</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QListWidget" name="tableListView">
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="columnListWidget">
   <property name="windowTitle">
    <string>Column List</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <property name="spacing">
        <number>0</number>
       </property>
       <item>
        <widget class="QLineEdit" name="columnListSearchEdit"/>
       </item>
       <item>
        <widget class="QPushButton" name="columnListSearchClear">
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QListWidget" name="columnListView">
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="resouces.qrc">
     <normaloff>:/rc/images/document-open.png</normaloff>:/rc/images/document-open.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="statusTip">
    <string>Open SQLite3 or text file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionInsert">
   <property name="icon">
    <iconset resource="resouces.qrc">
     <normaloff>:/rc/images/list-add.png</normaloff>:/rc/images/list-add.png</iconset>
   </property>
   <property name="text">
    <string>Insert</string>
   </property>
   <property name="statusTip">
    <string>Insert new row</string>
   </property>
   <property name="shortcut">
    <string>Ctrl++</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="icon">
    <iconset resource="resouces.qrc">
     <normaloff>:/rc/images/list-remove.png</normaloff>:/rc/images/list-remove.png</iconset>
   </property>
   <property name="text">
    <string>Delete</string>
   </property>
   <property name="statusTip">
    <string>Delete selected row</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+-</string>
   </property>
  </action>
  <action name="actionCommit">
   <property name="icon">
    <iconset resource="resouces.qrc">
     <normaloff>:/rc/images/document-save.png</normaloff>:/rc/images/document-save.png</iconset>
   </property>
   <property name="text">
    <string>Commit</string>
   </property>
   <property name="statusTip">
    <string>Commit chagnes</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="statusTip">
    <string>Quit Hyokai</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
   <property name="statusTip">
    <string>Close Window</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="resouces.qrc">
     <normaloff>:/rc/images/document-new.png</normaloff>:/rc/images/document-new.png</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="statusTip">
    <string>New SQLite3</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionCreateTable">
   <property name="icon">
    <iconset resource="resouces.qrc">
     <normaloff>:/rc/images/window-new.png</normaloff>:/rc/images/window-new.png</iconset>
   </property>
   <property name="text">
    <string>Create Table</string>
   </property>
   <property name="statusTip">
    <string>Create new table</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+N</string>
   </property>
  </action>
  <action name="actionAbout_Table_View">
   <property name="text">
    <string>About Hyokai</string>
   </property>
   <property name="statusTip">
    <string>About Hyokai</string>
   </property>
  </action>
  <action name="actionGo_github">
   <property name="text">
    <string>Go to github</string>
   </property>
   <property name="statusTip">
    <string>Go to github project page</string>
   </property>
  </action>
  <action name="actionImportTable">
   <property name="icon">
    <iconset resource="resouces.qrc">
     <normaloff>:/rc/images/toolbar/import.png</normaloff>:/rc/images/toolbar/import.png</iconset>
   </property>
   <property name="text">
    <string>Import Table</string>
   </property>
   <property name="statusTip">
    <string>Import Table from text or CSV</string>
   </property>
  </action>
  <action name="actionRevert">
   <property name="icon">
    <iconset resource="resouces.qrc">
     <normaloff>:/rc/images/edit-undo.png</normaloff>:/rc/images/edit-undo.png</iconset>
   </property>
   <property name="text">
    <string>Revert</string>
   </property>
   <property name="statusTip">
    <string>Revert changes</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="text">
    <string>About Qt</string>
   </property>
   <property name="statusTip">
    <string>About Qt</string>
   </property>
  </action>
  <action name="actionRun_Custum_SQL">
   <property name="icon">
    <iconset resource="resouces.qrc">
     <normaloff>:/rc/images/utilities-terminal.png</normaloff>:/rc/images/utilities-terminal.png</iconset>
   </property>
   <property name="text">
    <string>Run Custom SQL</string>
   </property>
   <property name="toolTip">
    <string>Run Custom SQL</string>
   </property>
   <property name="statusTip">
    <string>Run Custum SQL</string>
   </property>
  </action>
  <action name="actionRefresh">
   <property name="icon">
    <iconset resource="resouces.qrc">
     <normaloff>:/rc/images/view-refresh.png</normaloff>:/rc/images/view-refresh.png</iconset>
   </property>
   <property name="text">
    <string>Refresh</string>
   </property>
   <property name="statusTip">
    <string>Refresh table list and contents</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionExport_Table">
   <property name="icon">
    <iconset resource="resouces.qrc">
     <normaloff>:/rc/images/toolbar/export.png</normaloff>:/rc/images/toolbar/export.png</iconset>
   </property>
   <property name="text">
    <string>Export Table</string>
   </property>
   <property name="statusTip">
    <string>Export table as CSV or Tab separated text</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actionOpen_In_Memory_Database">
   <property name="text">
    <string>Open In Memory Database</string>
   </property>
   <property name="statusTip">
    <string>Open SQLite3 in memory database</string>
   </property>
  </action>
  <action name="actionAttach_Database">
   <property name="text">
    <string>Attach Database</string>
   </property>
   <property name="statusTip">
    <string>Attach exsists database</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="text">
    <string>Copy</string>
   </property>
   <property name="statusTip">
    <string>Copy contents</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionView_in_File_Manager">
   <property name="text">
    <string>Open File Location</string>
   </property>
   <property name="statusTip">
    <string>Open File Location with File Viewer</string>
   </property>
  </action>
  <action name="actionPreference">
   <property name="icon">
    <iconset resource="resouces.qrc">
     <normaloff>:/rc/images/preferences-system.png</normaloff>:/rc/images/preferences-system.png</iconset>
   </property>
   <property name="text">
    <string>Preference...</string>
   </property>
   <property name="statusTip">
    <string>Open Preference Pane</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+,</string>
   </property>
  </action>
  <action name="actionR_code_to_import">
   <property name="text">
    <string>Copy R code to import</string>
   </property>
   <property name="statusTip">
    <string>Generate R code to manipulate data with R </string>
   </property>
  </action>
  <action name="actionClear">
   <property name="text">
    <string>Clear</string>
   </property>
   <property name="statusTip">
    <string>Clear recent files</string>
   </property>
  </action>
  <action name="actionGo_to_SQLite3_webpage">
   <property name="text">
    <string>Go to SQLite3 webpage</string>
   </property>
  </action>
  <action name="actionDrop_Table">
   <property name="icon">
    <iconset resource="resouces.qrc">
     <normaloff>:/rc/images/user-trash.png</normaloff>:/rc/images/user-trash.png</iconset>
   </property>
   <property name="text">
    <string>Drop Table</string>
   </property>
   <property name="statusTip">
    <string>Drop this table or view</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Backspace</string>
   </property>
  </action>
  <action name="actionCopy_with_header">
   <property name="text">
    <string>Copy with header</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+C</string>
   </property>
  </action>
  <action name="actionSelect_All">
   <property name="text">
    <string>Select All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionConnect_to_database">
   <property name="text">
    <string>Connect to database...</string>
   </property>
  </action>
  <action name="actionDuplicate_connection">
   <property name="text">
    <string>Duplicate connection</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+D</string>
   </property>
  </action>
  <action name="actionDatabase_Information">
   <property name="text">
    <string>Database Information...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="actionShowDummyHiddenColumn">
   <property name="text">
    <string>Dummy</string>
   </property>
  </action>
  <action name="actionDuplicate_Table">
   <property name="text">
    <string>Duplicate Table...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionPlot">
   <property name="icon">
    <iconset resource="resouces.qrc">
     <normaloff>:/rc/images/x-office-spreadsheet.png</normaloff>:/rc/images/x-office-spreadsheet.png</iconset>
   </property>
   <property name="text">
    <string>Plot...</string>
   </property>
   <property name="toolTip">
    <string>Plot chart</string>
   </property>
  </action>
  <action name="actionDummy">
   <property name="text">
    <string>Dummy</string>
   </property>
  </action>
  <action name="actionGo_to_Hyokai_info">
   <property name="text">
    <string>Go to Hyokai.info</string>
   </property>
  </action>
  <action name="actionUse_fixed_width_font">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Use fixed width font</string>
   </property>
  </action>
  <action name="actionShow_Toolbar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Toolbar</string>
   </property>
  </action>
  <action name="actionShow_Table_List_View">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Table List View</string>
   </property>
  </action>
  <action name="actionShow_Column_List_View">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Column List View</string>
   </property>
  </action>
  <action name="actionSplit_Window">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Split Window</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>SQLTextEdit</class>
   <extends>QPlainTextEdit</extends>
   <header>sqltextedit.h</header>
  </customwidget>
  <customwidget>
   <class>QTabBar</class>
   <extends>QWidget</extends>
   <header>QTabBar</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resouces.qrc"/>
 </resources>
 <connections/>
</ui>
